cupy/core/core.pyx

```diff
@@ -1717,51 +1719,213 @@ cpdef ndarray array(obj, dtype=None, bint copy=True, order='K',
     elif hasattr(obj, '__cuda_array_interface__'):
         return array(_convert_object_with_cuda_array_interface(obj),
                      dtype, copy, order, subok, ndmin)
+    else:  # obj is sequence, numpy array, scalar or the other type of object
+        shape, elem_type, elem_dtype = _get_concat_shape(obj)
+        if shape is not None and shape[-1] != 0:
+            # obj is a non-empty sequence of ndarrays which share same shape
+            # and dtype
+
+            # resulting array is C order unless 'F' is explicitly specified
+            # (i.e., it ignores order of element arrays in the sequence)
+            order = (
+                'F'
+                if order is not None and len(order) >= 1 and order[0] in 'Ff'
+                else 'C')
+            ndim = len(shape)
+            if ndmin > ndim:
+                shape = (1,) * (ndmin - ndim) + shape
+
+            if dtype is None:
+                dtype = elem_dtype
+            # Note: dtype might not be numpy.dtype in this place
+
+            if issubclass(elem_type, numpy.ndarray):
+                # obj is Seq[numpy.ndarray]
+                a = _send_numpy_array_list_to_gpu(
+                    obj, elem_dtype, dtype, shape, order, ndmin)
+            elif issubclass(elem_type, ndarray):
+                # obj is Seq[cupy.ndarray]
+                lst = _flatten_list(obj)
+                if len(shape) == 1:
+                    # convert each scalar (0-dim) ndarray to 1-dim
+                    lst = [cupy.expand_dims(x, 0) for x in lst]
+
+                a = (_manipulation.concatenate_method(lst, 0)
+                                  .reshape(shape)
+                                  .astype(dtype, order=order, copy=False))
+            else:  # should not be reached here
```

Comment:
I think it **can** reach here depending on the input.
Please remove this comment.

Reply:
`isinstance(elem_type, (numpy.ndarray, cupy.ndarray))` is always true when `shape != None`. If the user sees this exception, it must be a bug of `_get_concat_shape()`.

Reply:
Then, please use `assert` instead of exception.